src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/87.diff.sql

```diff
@@ -0,0 +1,49 @@
+ALTER PROCEDURE dbo.MergeResourcesBeginTransaction @Count int, @TransactionId bigint OUT, @SequenceRangeFirstValue int = NULL OUT, @HeartbeatDate datetime = NULL
+AS
+set nocount on
+DECLARE @SP varchar(100) = 'MergeResourcesBeginTransaction'
+       ,@Mode varchar(200) = 'Cnt='+convert(varchar,@Count)+' HB='+isnull(convert(varchar,@HeartbeatDate,121),'NULL')
+       ,@st datetime = getUTCdate()
+       ,@FirstValueVar sql_variant
+       ,@LastValueVar sql_variant
+       -- Optimal concurrency slowly increases with transaction size growth. For single resource transactions optimal value is ~64.
+       -- For large transactions (1000 resources) typical for $import it is ~256. 
+       -- Transaction size is hard to get in efficient way, so I choose 256 as default.
+       ,@OptimalConcurrency int = isnull((SELECT Number FROM Parameters WHERE Id = 'MergeResources.OptimalConcurrentCalls'), 256)
+       ,@CurrentConcurrency int
+       ,@msg varchar(1000)
+
+BEGIN TRY
+  SET @TransactionId = NULL
+
+  IF @@trancount > 0 RAISERROR('MergeResourcesBeginTransaction cannot be called inside outer transaction.', 18, 127)
+
+  SET @CurrentConcurrency = (SELECT count(*) FROM sys.dm_exec_sessions WHERE status <> 'sleeping' AND program_name = 'MergeResources')
+  IF @CurrentConcurrency > @OptimalConcurrency
+  BEGIN
+    SET @msg = 'Number of concurrent MergeResources calls = '+convert(varchar,@CurrentConcurrency)+' is above optimal = '+convert(varchar,@OptimalConcurrency)+'.';
+    THROW 50410, @msg, 1 
+  END
+
+  SET @FirstValueVar = NULL
+  WHILE @FirstValueVar IS NULL
+  BEGIN
+    EXECUTE sys.sp_sequence_get_range @sequence_name = 'dbo.ResourceSurrogateIdUniquifierSequence', @range_size = @Count, @range_first_value = @FirstValueVar OUT, @range_last_value = @LastValueVar OUT
+    SET @SequenceRangeFirstValue = convert(int,@FirstValueVar)
+    IF @SequenceRangeFirstValue > convert(int,@LastValueVar)
+      SET @FirstValueVar = NULL
+  END
+
+  SET @TransactionId = datediff_big(millisecond,'0001-01-01',sysUTCdatetime()) * 80000 + @SequenceRangeFirstValue
+
+  INSERT INTO dbo.Transactions
+         (  SurrogateIdRangeFirstValue,   SurrogateIdRangeLastValue,                      HeartbeatDate )
```

Comment:
I see that line 40 and 41 have these continuous spaces ("                      "). Is that a formatting issue?

Reply:
This is because INSERT column list is aligned with SELECT list.